### PR TITLE
BITAU-90 Add "Sync with Bitwarden App" row to settings

### PR DESCRIPTION
**Pull request by ahaisting-livefront**
## 🎟️ Tracking

https://livefront.atlassian.net/browse/BITAU-90

## 📔 Objective

The goal of this PR is to add a row to that takes the user to the Bitwarden account security screen so that they can turn on account syncing. If the user doesn't have Bitwarden installed, it should take them to the play store listing. 

## 📸 Screenshots

<video src="https://github.com/user-attachments/assets/1c86bc2c-16f3-4cc1-89ac-4cecadf88400" width="300" />

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/main/kotlin/com/bitwarden/authenticator/ui/platform/feature/settings/SettingsScreen.kt [hunk 1]
```diff
@@ -150,6 +169,12 @@ fun SettingsScreen(
                         viewModel.trySendAction(SettingsAction.DataClick.BackupClick)
                     }
                 },
+                onSyncWithBitwardenClick = remember(viewModel) {
+                    {
+                        viewModel.trySendAction(SettingsAction.DataClick.SyncWithBitwardenClick)
```
Comment (SaintPatrck) [inline app/src/main/kotlin/com/bitwarden/authenticator/ui/platform/feature/settings/SettingsScreen.kt:174]: Can we get a test for this click event?

File: app/src/main/kotlin/com/bitwarden/authenticator/ui/platform/feature/settings/SettingsScreen.kt [hunk 2]
```diff
@@ -294,6 +322,25 @@ private fun VaultSettings(
         dialogConfirmButtonText = stringResource(R.string.learn_more),
         dialogDismissButtonText = stringResource(R.string.ok),
     )
+    if (shouldShowSyncWithBitwardenApp) {
```
Comment (SaintPatrck) [inline app/src/main/kotlin/com/bitwarden/authenticator/ui/platform/feature/settings/SettingsScreen.kt:325]: Can we get a test to validate this is shown/hidden based on state?

Comment (ahaisting-livefront) [reply app/src/main/kotlin/com/bitwarden/authenticator/ui/platform/feature/settings/SettingsScreen.kt:325]: Added :)

File: app/src/main/kotlin/com/bitwarden/authenticator/ui/platform/feature/settings/SettingsScreen.kt (hunk 1 above)
Comment (ahaisting-livefront) [reply app/src/main/kotlin/com/bitwarden/authenticator/ui/platform/feature/settings/SettingsScreen.kt:174]: Yep! I added tests for handling each of the new events as well.

File: fastlane/Fastfile [hunk 3]
```diff
@@ -19,7 +19,7 @@ platform :android do
 
   desc "Runs tests"
   lane :check do
-    gradle(tasks: ["check","koverXmlReportDebug"])
+    gradle(tasks: ["testDebug", "lintDebug", "detekt","koverXmlReportDebug"])
```
Comment (ahaisting-livefront) [inline fastlane/Fastfile:22]: I made a separate PR for this change:

https://github.com/bitwarden/authenticator-android/pull/242

**Review by SaintPatrck — APPROVED** [review]

**Comment by ahaisting-livefront** [comment]
Thanks, @SaintPatrck !

**Pull request merged**